tests/test_resources/test_modules/test_functions/test_function.py

```diff
@@ -173,301 +75,469 @@ async def async_slow_generator(size):
+    @pytest.mark.level("local")
+    def test_function_queueing(self, cluster):
+        pid_fn = rh.function(getpid).to(cluster)
+
+        pids = [pid_fn.enqueue(resources={"num_cpus": 1}) for _ in range(10)]
+        assert len(pids) == 10
+
+    def test_function_to_env(self, cluster):
+        cluster.run(["pip uninstall numpy -y"])
+
+        np_func = rh.function(np_array).to(cluster, env=["numpy"])
+
+        list = [1, 2, 3]
+        res = np_func(list)
+        assert res.tolist() == list
+
+    @unittest.skip("Not working properly.")
+    # originally used ondemand_cpu_cluster, therefore marked as minimal
+    @pytest.mark.level("minimal")
+    def test_function_external_fn(self, cluster):
+        """Test functioning a module from reqs, not from working_dir"""
+        import torch
+
+        re_fn = rh.function(torch.sum).to(cluster, env=["torch"])
+        res = re_fn(torch.arange(5))
+        assert int(res) == 10
+
+    @unittest.skip("Runs indefinitely.")
+    # originally used ondemand_cpu_cluster, therefore marked as minimal
+    @pytest.mark.level("minimal")
+    def test_notebook(self, cluster):
+        nb_sum = lambda x: multiproc_torch_sum(x)
+        re_fn = rh.function(nb_sum).to(cluster, env=["torch==1.12.1"])
+
+        re_fn.notebook()
+        summands = list(zip(range(5), range(4, 9)))
+        res = re_fn(summands)
+
+        assert res == [4, 6, 8, 10, 12]
+        re_fn.delete_configs()
+
+    @unittest.skip("Runs indefinitely.")
+    def test_ssh(self):
+        # TODO do this properly
+        my_function = rh.function(name="local_function")
+        my_function.ssh()
+        assert True
+
+    @pytest.mark.rnstest
+    @pytest.mark.level("local")
+    def test_share_function(self, cluster):
+        my_function = rh.function(fn=summer).to(cluster).save(REMOTE_FUNC_NAME)
+
+        my_function.share(
+            users=["donny@run.house", "josh@run.house"],
+            access_type="read",
+            notify_users=False,
+        )
+        assert True
+
+    @pytest.mark.rnstest
+    @pytest.mark.level("local")
+    def test_load_shared_function(self):
+        my_function = rh.function(name=REMOTE_FUNC_NAME)
+        res = my_function(1, 2)
+        assert res == 3
+
+    @pytest.mark.rnstest
+    @pytest.mark.level("local")
+    def delete_function_from_rns(self, s):
+        server_url = s.rns_client.api_server_url
+        resource_request_uri = s.rns_client.resource_uri(s.name)
+        resp = requests.delete(
+            f"{server_url}/resource/{resource_request_uri}",
+            headers=s.rns_client.request_headers,
+        )
+        if resp.status_code != 200:
+            raise Exception(
+                f"Failed to delete_configs function data from path: {load_resp_content(resp)}"
+            )
+
+        try:
+            # Terminate the cluster
+            s.cluster.teardown_and_delete()
+        except Exception as e:
+            raise Exception(f"Failed to teardown the cluster: {e}")
```

Comment:
I don't think we need this anymore, it's being tested at the resource level.

Reply:
Indeed, seems like there a similar test flow in a test called test_save_and_load in resource test. Removed.